pkg/services/cleanup/cleanup.go

```diff
@@ -418,3 +425,133 @@ func (srv *CleanUpService) cleanUpTrashAlertRules(ctx context.Context) {
 		logger.Debug("Cleaned up deleted alert rules", "rows affected", affected)
 	}
 }
+
+// cleanupStaleLBACRules exists to clean up lbac rules that are stale from teams getting deleted as we do not have
+// cascading deletions on teams to delete existing lbac rules
+func (srv *CleanUpService) cleanupStaleLBACRules(ctx context.Context) {
+	logger := srv.log.FromContext(ctx)
+
+	// Get all datasources
+	allDataSources, err := srv.dataSourceService.GetAllDataSources(ctx, &datasources.GetAllDataSourcesQuery{})
+	if err != nil {
+		logger.Error("Failed to get datasources for LBAC cleanup", "error", err)
+		return
+	}
+
+	var totalCleaned int
+	var totalDataSources int
+
+	for _, ds := range allDataSources {
+		if ds.JsonData == nil {
+			continue
+		}
+
+		// Check if datasource has team LBAC rules
+		teamHTTPHeaders, err := datasources.GetTeamHTTPHeaders(ds.JsonData)
+		if err != nil || teamHTTPHeaders == nil {
+			continue
```

Comment:
We may want to know if parsing the headers fail so we can investigate and handle whatever failure is happening.  I think this fits into the spirit of this function as well... cleaning stale (or in this case, unparseable) rules.

To that end, maybe log something here?

Reply:
yea, we could add some logging here